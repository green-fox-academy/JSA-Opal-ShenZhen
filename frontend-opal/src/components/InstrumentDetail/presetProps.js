import PropTypes from 'prop-types';

const defaultDetailData = [];
const defaultChartData = [];
const detailDataTypes = PropTypes.arrayOf(
  PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    value: PropTypes.string,
    group: PropTypes.number
  })
);
const chartDataTypes = PropTypes.arrayOf(
  PropTypes.shape({
    x: PropTypes.string,
    y: PropTypes.number
  })
);

export default {
  defaultDetailData,
  defaultChartData,
  detailDataTypes,
  chartDataTypes
};
