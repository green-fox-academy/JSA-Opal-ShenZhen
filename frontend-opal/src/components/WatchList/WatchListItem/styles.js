import { StyleSheet } from 'react-native';

import tools from 'components/common/tools';

export default StyleSheet.create({
  container: {
    marginBottom: tools.getPixel(20)
  }
});
