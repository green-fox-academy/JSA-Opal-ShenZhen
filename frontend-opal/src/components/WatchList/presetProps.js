import PropTypes from 'prop-types';

const defaultQuote = {
  symbol: '',
  latestPrice: 0,
  changePercent: '',
  marketCap: '',
  week52High: 0,
  week52Low: 0,
  peRatio: 0,
  eps: '',
  divYield: '',
  chartData: []
};

const quoteTypes = PropTypes.shape({
  symbol: PropTypes.string,
  latestPrice: PropTypes.number,
  changePercent: PropTypes.string,
  marketCap: PropTypes.string,
  week52High: PropTypes.number,
  week52Low: PropTypes.number,
  peRatio: PropTypes.number,
  eps: PropTypes.string,
  divYield: PropTypes.string,
  chartData: PropTypes.arrayOf(
    PropTypes.shape({
      x: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      y: PropTypes.number
    })
  )
});

export default {
  defaultQuote,
  quoteTypes
};
