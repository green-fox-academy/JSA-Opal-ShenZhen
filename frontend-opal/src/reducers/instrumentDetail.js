const pairs = {
  GET_INSTRUMENT_DETAIL_DATA: (state, action) => action.payload.data,
  CLEAR_DETAIL_DATA: (state, action) => []
};

export default pairs;
