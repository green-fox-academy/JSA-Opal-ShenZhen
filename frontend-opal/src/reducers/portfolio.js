const pairs = {
  GET_PORTFOLIO_DATA: (state, action) => {
    return [...state, action.payload.data];
  }
};

export default pairs;
