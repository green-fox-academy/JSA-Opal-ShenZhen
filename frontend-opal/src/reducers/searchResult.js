const pairs = new Map();
pairs.set('SEARCH', search);

function searchResult(state = {}, action) {
  if (pairs.has(action.type)) return pairs.get(action.type)(state, action);
  return state;
}

function search(state, action) {
  return action.payload.results;
}

export default searchResult;
