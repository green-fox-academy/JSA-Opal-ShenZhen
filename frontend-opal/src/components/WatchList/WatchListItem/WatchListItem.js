import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';

import SingleInstrument from 'components/WatchList/SingleInstrument/SingleInstrument';
import presetProps from 'components/WatchList/presetProps';

const WatchListItem = ({ data, index }) => {
  return (
    <View>
      {data.map(info => (
        <View key={`${info.symbol} item`}>
          <SingleInstrument data={info} index={index} />
        </View>
      ))}
    </View>
  );
};

WatchListItem.defaultProps = {
  data: [presetProps.defaultQuote],
  index: 0
};

WatchListItem.propTypes = {
  data: PropTypes.arrayOf(presetProps.quoteTypes),
  index: PropTypes.number
};

export default WatchListItem;
