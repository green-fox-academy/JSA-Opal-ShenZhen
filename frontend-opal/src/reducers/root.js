import { combineReducers } from 'redux';

import watchLists from './Watchlists';
import searchResult from './SearchResult';

export default combineReducers({
  watchLists,
  searchResult,
});
