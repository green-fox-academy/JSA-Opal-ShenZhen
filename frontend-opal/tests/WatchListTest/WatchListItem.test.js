import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducers from 'reducers/root';
import WatchListItem from 'components/WatchList/WatchListItem/WatchListItem';
import presetProps from 'components/WatchList/presetProps';

describe('<WatchListItem />', () => {
  it('renders correctly', () => {
    const info = {
      item: '',
      data: [presetProps.defaultQuote]
    };
    const tree = renderer.create(<WatchListItem item={info.item} data={info.data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
