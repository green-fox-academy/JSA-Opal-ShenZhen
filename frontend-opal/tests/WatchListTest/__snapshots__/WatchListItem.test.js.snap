// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WatchListItem /> renders correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "black",
        "borderStyle": "solid",
        "borderWidth": 2,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
      }
    }
  >
    <Text>
      
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
